### PR TITLE
Make CLI and API consistent for reports

### PR DESCRIPTION
**Pull request by kholdaway**

### CODE REVIEW

File: qpc/report/deployments.py [hunk 1]
```diff
@@ -27,14 +27,14 @@
 
 
 # pylint: disable=too-few-public-methods
-class ReportSummaryCommand(CliCommand):
+class ReportDeploymentsCommand(CliCommand):
     """Defines the report summary command.
```
Comment (myersCody) [inline qpc/report/deployments.py]: You missed one. summary => Deployments

Comment (myersCody) [inline qpc/report/deployments.py]: You missed one

File: qpc/messages.py [hunk 2]
```diff
@@ -194,9 +199,9 @@
 REPORT_SJ_DOES_NOT_EXIST = \
     'Scan Job %s does not exist.'
 REPORT_SJS_DO_NOT_EXIST = 'The following scan jobs do not exist: %s.'
-REPORT_NO_SUMMARY_REPORT_FOR_SJ = \
+REPORT_NO_DEPLOYMENTS_REPORT_FOR_SJ = \
```
Comment (abaiken) [inline qpc/messages.py]: Not a blocker but could we possibly combine these two error messages and just add the "scan job/report job" when we are calling the message?

File: qpc/report/__init__.py [hunk 3]
```diff
@@ -12,8 +12,15 @@
 """Constants for the Report commands."""
 
 SUBCOMMAND = 'report'
+
+
```
Comment (abaiken) [inline qpc/report/__init__.py]: Are two spaces necessary here?

File: qpc/report/deployments.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 # along with this software; if not, see
 # https://www.gnu.org/licenses/gpl-3.0.txt.
 #
-"""ReportSummaryCommand is used to show summary report information."""
+"""ReportDeploymentsCommand is used to show summary report information."""
```
Comment (abaiken) [inline qpc/report/deployments.py]: change this docstring to reflect deployments report information

File: qpc/report/details.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 # along with this software; if not, see
 # https://www.gnu.org/licenses/gpl-3.0.txt.
 #
-"""ReportDetailCommand is used to show detail report information."""
+"""ReportDetailsCommand is used to show detail report information."""
```
Comment (abaiken) [inline qpc/report/details.py]: change `detail` to `details`

File: qpc/report/tests_deprecated_report_detail.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+from qpc.utils import get_server_location, write_server_config
+
+import requests_mock
+
+PARSER = ArgumentParser()
+SUBPARSER = PARSER.add_subparsers(dest='subcommand')
+
+
+class ReportDetailTests(unittest.TestCase):
+    """Class for testing the scan show commands for qpc."""
```
Comment (abaiken) [inline qpc/report/tests_deprecated_report_detail.py]: I know that this was not introduced by you but change this docstring to reflect report commands

File: qpc/report/tests_report_deployments.py [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+from qpc.utils import get_server_location, write_server_config
+
+import requests_mock
+
+PARSER = ArgumentParser()
+SUBPARSER = PARSER.add_subparsers(dest='subcommand')
+
+
+class ReportDeploymentsTests(unittest.TestCase):
+    """Class for testing the scan show commands for qpc."""
```
Comment (abaiken) [inline qpc/report/tests_report_deployments.py]: same command as above

File: qpc/report/deployments.py (hunk 1 above)
Comment (kholdaway) [reply qpc/report/deployments.py]: 👍

File: qpc/report/__init__.py (hunk 3 above)
Comment (kholdaway) [reply qpc/report/__init__.py]: 👍

File: qpc/report/deployments.py (hunk 4 above)
Comment (kholdaway) [reply qpc/report/deployments.py]: 👍

File: qpc/report/details.py (hunk 5 above)
Comment (kholdaway) [reply qpc/report/details.py]: 👍

File: qpc/report/tests_deprecated_report_detail.py (hunk 6 above)
Comment (kholdaway) [reply qpc/report/tests_deprecated_report_detail.py]: 😂

File: qpc/report/tests_report_deployments.py (hunk 7 above)
Comment (kholdaway) [reply qpc/report/tests_report_deployments.py]: classic

File: docs/source/working_with_reports.rst [hunk 8]
```diff
@@ -32,4 +32,12 @@ Additionally, if you would like to create a merged report from JSON details repo
 
   # qpc report merge --json-files /path/to/file1.JSON /path/to/file2.JSON
 
-The output of the above commands provides a report identifier that can be used to access the merged data. You can use this identifier and the ``qpc report`` command with the ``detail`` or ``summary`` subcommands to generate a report from the merged results.
+You may also specify a directory with many JSON details reports with the following command::
+
+  # qpc report merge --json-directory /path/to/directory_with_json
+
+The above commands run an asynchronous job.  The output of the above commands provides a job id that can be used to check the status of the merge job.  To check the status of a merge job, run the following command::
```
Comment (abaiken) [inline docs/source/working_with_reports.rst]: change `commands` to `command`

File: docs/source/working_with_reports.rst [hunk 9]
```diff
@@ -32,4 +32,12 @@ Additionally, if you would like to create a merged report from JSON details repo
...
+
+# qpc report merge-status --job 1
+
+The output of the above commands provides a report identifier that can be used to access the merged data. You can use this identifier and the ``qpc report`` command with the ``details`` or ``deployments`` subcommands to generate a report from the merged results.
```
Comment (abaiken) [inline docs/source/working_with_reports.rst]: same comment as above

File: docs/source/man.rst [hunk 10]
```diff
@@ -551,7 +551,7 @@ For example, the raw facts of a scan that includes both network and vcenter sour
 
 Merging Scan Job Results
 ~~~~~~~~~~~~~~~~~~~~~~~~
-The ``qpc report merge`` command combines results from two or more scan job identifiers, report identifers, or JSON details report files to create a single report. The ``qpc report summary`` or ``qpc report detail`` can be used to obtain the report.
+The ``qpc report merge`` command combines results from two or more scan job identifiers, report identifers, or JSON details report files to create a single report. The ``qpc report deployments`` or ``qpc report details`` can be used to obtain the report.
```
Comment (abaiken) [inline docs/source/man.rst]: change the last sentence to `The ``qpc report merge`` command returns the identifier of the report that is created. You can use this identifier and the ``qpc report`` command with the ``details`` or ``deployments`` subcommands to generate a report from the merged results.`

File: docs/source/man.rst [hunk 11]
```diff
@@ -492,14 +492,14 @@ The ``qpc scan cancel`` command cancels the execution of a scan job. A canceled
...
+Use the ``qpc report`` command to generate a report from a scan. You can generate a report as JavaScript Object Notation (JSON) format or as comma-separated values (CSV) format. There are two different types of report that you can generate, a *details* report and a *deployments* report.
 
 
-Viewing the Detail Report
+Viewing the Details Report
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-The ``qpc report detail`` command generates a detailed report that contains the unprocessed facts that are gathered during a scan. These facts are the raw output from network, vcenter, and satellite scans, as applicable.
+The ``qpc report details`` command generates a detailed report that contains the unprocessed facts that are gathered during a scan. These facts are the raw output from network, vcenter, and satellite scans, as applicable.  The ``qpc report detail`` commands is now deprecated.  The ``qpc report details`` command should be used instead.
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply docs/source/man.rst]: 👍

File: docs/source/man.rst [hunk 12]
```diff
@@ -521,13 +521,13 @@ The ``qpc report detail`` command generates a detailed report that contains the
 
   Required. Path to a file location where the report data is saved.
 
-Viewing the Summary Report
+Viewing the Deployments Report
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
-The ``qpc report summary`` command generates a report that contains the processed fingerprints from a scan. A *fingerprint* is the set of system, product, and entitlement facts for a particular physical or virtual machine. A processed fingerprint results from a procedure that merges facts from various sources, and, when possible, deduplicates redundant systems.
+The ``qpc report deployments`` command generates a report that contains the processed fingerprints from a scan. A *fingerprint* is the set of system, product, and entitlement facts for a particular physical or virtual machine. A processed fingerprint results from a procedure that merges facts from various sources, and, when possible, deduplicates redundant systems.  The ``qpc report summary`` commands is now deprecated.  The ``qpc report deployments`` command should be used instead.
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply docs/source/man.rst]: 👍

File: docs/source/quick_start.rst [hunk 13]
```diff
@@ -98,16 +98,16 @@ When you run the ``scan start`` command, the output provides an identifier for t
 
 Viewing the Scan Report
 -----------------------
-When the scan job completes, you have the capability to produce a report for that scan job. You can request a report with all the details, or facts, of the scan, or request a report with a summary. The summary report process runs steps to merge the facts found during the inspection of the various hosts that are contacted during the scan. When possible, the report process also runs steps to deduplicate redundant systems. For both types of reports, you can produce the report in JavaScript Object Notation (JSON) format or comma-separated values (CSV) format.
+When the scan job completes, you have the capability to produce a report for that scan job. You can request a report with all the details, or facts, of the scan, or request a report deployments. The deployments report process runs steps to merge the facts found during the inspection of the various hosts that are contacted during the scan. When possible, the report process also runs steps to deduplicate redundant systems. For both types of reports, you can produce the report in JavaScript Object Notation (JSON) format or comma-separated values (CSV) format.
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply docs/source/quick_start.rst]: 👍

File: docs/source/requirements.rst [hunk 14]
```diff
@@ -6,6 +6,6 @@ Before installing Quipucords in your environment, review the following guideline
 - The system that Quipucords is installed on must have access to the systems to be discovered and inspected.
 - Any network sources that are targeted for the inspection process must be running SSH.
 - For a scan of network systems, the user account (credential) that Quipucords uses for the SSH connection into the target systems must have adequate permissions to run commands and read certain files on those systems. For example, some of the commands require privilege elevation to gather the complete set of facts for the scan. For more information about these commands, see `Commands Used in Scans of Remote Network Assets <commands.html>`_
-- The credential user account requires an ``sh`` shell or a similar shell. For example, the shell *cannot* be the ``/sbin/nologin`` or ``/bin/false`` shell.
+- The credential user account requires an ``bash`` shell. For example, the shell *cannot* be the ``/sbin/nologin`` or ``/bin/false`` shell.
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply docs/source/requirements.rst]: 👍

File: docs/source/working_with_reports.rst (hunk 8 above)
Comment (kholdaway) [reply docs/source/working_with_reports.rst]: 👍

File: docs/source/working_with_reports.rst (hunk 9 above)
Comment (kholdaway) [reply docs/source/working_with_reports.rst]: 👍

File: docs/source/man.rst (hunk 10 above)
Comment (kholdaway) [reply docs/source/man.rst]: 👍

File: docs/source/man.rst [hunk 15]
```diff
@@ -492,14 +492,14 @@ The ``qpc scan cancel`` command cancels the execution of a scan job. A canceled
...
+Use the ``qpc report`` command to generate a report from a scan. You can generate a report as JavaScript Object Notation (JSON) format or as comma-separated values (CSV) format. There are two different types of report that you can generate, a *details* report and a *deployments* report.
 
 
-Viewing the Detail Report
+Viewing the Details Report
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-The ``qpc report detail`` command generates a detailed report that contains the unprocessed facts that are gathered during a scan. These facts are the raw output from network, vcenter, and satellite scans, as applicable.
+The ``qpc report details`` command generates a detailed report that contains the unprocessed facts that are gathered during a scan. These facts are the raw output from network, vcenter, and satellite scans, as applicable.  The ``qpc report details`` command is now deprecated.  The ``qpc report details`` command should be used instead.
```
Comment (abaiken) [inline docs/source/man.rst]: you should change the first ``qpc report details`` back to ``qpc report detail``

Comment (kholdaway) [reply docs/source/man.rst]: 😂 Indeed.

File: qpc.spec [hunk 16]
```diff
@@ -56,10 +56,10 @@ install -D -p -m 644 build/qpc.1 $RPM_BUILD_ROOT%{_mandir}/man1/qpc.1
 %{_mandir}/man1/qpc.1.gz
 
 %changelog
-* Mon Oct 8 2018 Ashley Aiken <aaiken@redhat.com> 0.0.45-1
-- Add server status command to QPC
-* Fri Oct 5 2018 Cody Myers <cmyers@redhat.com> 0.0.45-1
-- Add async merge job creation and job status
+* Fri Oct 19 2018 Cody Myers <cmyers@redhat.com> 0.0.45-1
```
Comment (abaiken) [inline qpc.spec]: Weird. Didn't know your name was Cody Myers 😂

**Pull request merged**